### ISSUE TITLE
OnlineDataset Issues

### ISSUE THREAD

**Issue by urimerhav** [hkchengrex/CascadePSP#17]
1. perturb=False flag is broken (`self.bilinear_dual_transform_im is missing`)
2. Much more worrying: the code seems to be applying horizontal flip independently for ground truth and label. With 50% chance, that means the ground truth is corrupted and no longer aligns with the rgb image.

**Comment by hkchengrex** [comment]
Thanks for reporting.

For 2 it is probably due to the new torchvision version, see https://github.com/pytorch/vision/issues/2520.
One quick fix is to revert to torchvision 0.6.0.

I will probably fix these soon.

**Comment by urimerhav** [comment]
Got it. 

By the way, it looks like the way the code handles the dataset, you look at random crops of 224 pixels. This is surprising because the meaning of 224 pixels may be either very broad context if the original image is pretty small, or a tiny patch takes completely out of context if the original image was, say, 4000X4000 pixels. Is this intended behavior?

**Comment by hkchengrex** [comment]
It is intended. Note that one of our key features is that we only need to train on low-resolution images with the largest being around 400x500 while extending to high-resolution only in evaluation. You should not be using the online dataset for large images. If you have sufficiently many annotated high-resolution training images, this work is probably not the most suitable (GLNet might be better in that situation).

**Comment by urimerhav** [comment]
Right. It's certainly counter-intuitive that training on one scale and predicting on another would yield such impressive results! I guess I need to reread what you wrote on this in more depth.

**Comment by hkchengrex** [comment]
Fixed.
